manager/filters.py

```diff
@@ -145,11 +145,39 @@ def _filter_cve_by_show_all(query, args):
     Returns:
         object: Modified query with show all filter applied
     """
+    if args['affecting'] is not None:
+        return query
     if 'show_all' not in args or not args['show_all']:
         query = query.where(CveAccountData.systems_affected > 0)
     return query
 
 
+def _filter_cve_by_affecting(query, args):
+    """
+    Filters list of CVEs based on the fact whether CVE applies at least to one system or not. Query has to contain CveAccountData table.
+
+    Args:
+        query (object): Query object to apply filters to
+        args (dict): Query arguemnts
+
+    Returns:
+        object: Modified query with affecting filter applied
+    """
+    if args['show_all'] is not None:
+        return query
+
+    expr = [True, False] if args['affecting'] is None else args['affecting']
+
+    if expr == [True, False]:
+        query = query.where(CveAccountData.systems_affected > 0)
+    elif expr == [False, True]:
+        query = query.where(CveAccountData.systems_affected.is_null(True))
+    elif expr == [False, False]:
+        query = query.where(CveAccountData.systems_affected == 0)
```

Comment:
is this correct? shouldn't it be for [False, True] option?

Reply:
This is to force empty list when no checkbox is ticked. Seems logical to me with the default state of UI where 1st checkbox will always be ticked when you come to the CVE page.

Reply:
But what if there is row in CveAccountData with systems_affected = 0? For example with set status on CVE?

Reply:
@jdobes Good catch, I'll change it to something like < -10000.